### PR TITLE
Facet UI changes to limit number of checked items

### PR DESCRIPTION
**Pull request by Akrion**
- Sets a soft limit of 250 checked items at which point we display a note to the user:

CleanShot 2021-07-01 at 22 35 52@2x

- `Select All` is hidden if the visible items are more than 500. `Select All` is visible if all the items are checked to allow the user to uncheck all

This currently does not change the old searches values to only send 500. My thought process here was to limit that at the CES if we want to.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: e2c9dc63ef569a5767bd3069b4dc587feebed6eb


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2ac5ca1116d432a3f8038686f060eb19fd28addc
</p>

File: src/utils/facet.js [hunk 1]
```diff
@@ -21,61 +21,67 @@ export let displayBlankFn = () => <i>Not Specified</i>
...
-      justifyContent="space-between"
-    >
-      <div>
-        Showing{' '}
-        {toNumber(_.min([node.size || 10, _.size(node.context.options)]))} of{' '}
-        {toNumber(node.context.cardinality)}
-      </div>
-      {node.context.cardinality > (node.size || 10) && (
+)(({ node, tree }) => {
+  let size = _.getOr(10, 'size', node)
```
Comment (daedalus28) [inline src/utils/facet.js:25]: This is a slightly different behavior - before, size would be 10 if it was set to 0, but this will change the behavior when you pass in 0 on purpose. Was this change intentional?

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: src/utils/facet.js (hunk 1 above)
Comment (Akrion) [reply src/utils/facet.js]: PAssing size 0 on purpose did not make sense to me... I changed it back.

File: src/utils/facet.js [hunk 2]
```diff
@@ -21,61 +21,67 @@ export let displayBlankFn = () => <i>Not Specified</i>
...
   )
-  let allSelected = _.isEmpty(missingOptions)
-  return (
+  let isAllChecked = _.isEmpty(notChecked)
+  let allChecked = [...node.values, ...notChecked]
+  let isOverTheLimit = _.size(allChecked) > maxChecked
+
+  // If the items are all already selected and we are not going to be over the max if we select all
+  // then show the "Select All". This way we still allow the user to be able to "Unselect all"
+  return !isOverTheLimit || isAllChecked ? (
```
Comment (stellarhoof) [inline src/utils/facet.js:68]: I think we can hide it even if they're all checked, since clearing the filter is equivalent to unselecting all.

**Review by stellarhoof — APPROVED** [review]
